### PR TITLE
FeatureTE7-T1127-web-services

### PR DESCRIPTION
**Pull request by SagarAjudiya**
- URLSession
- GET, POST
- UploadTask
- DownLoadTask
- MVVM
- Coordinates

### CODE REVIEW

File: UIComponents/UIComponents.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -109,36 +102,16 @@
 		1BB03C452A1CE2EF00AAC4B1 /* JobHomeScreenCell.xib in Resources */ = {isa = PBXBuildFile; fileRef = 1BB03C442A1CE2EF00AAC4B1 /* JobHomeScreenCell.xib */; };
 		1BB03C4C2A1CEF3000AAC4B1 /* ScrollView.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 1BB03C4B2A1CEF3000AAC4B1 /* ScrollView.storyboard */; };
 		1BB03C4E2A1CEF5200AAC4B1 /* ScrollViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1BB03C4D2A1CEF5200AAC4B1 /* ScrollViewController.swift */; };
+		1BB58BD02A4D41D40046E685 /* R.generated.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1BB58BCF2A4D41D40046E685 /* R.generated.swift */; };
```
Comment (Shilpriya-Simform) [inline UIComponents/UIComponents.xcodeproj/project.pbxproj:105]: R.generated.swift should not be on PR

File: UIComponents/UIComponents/WebServices/Module/Upload/Controller/MethodsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        pvLine.progress = 0
+        avIndicator.isHidden = true
+        lblUploadProgress.text = ""
+    }
+    
+    // MARK: Bind ViewModel
+    private func bindViewModel() {
+        viewModel.uploadProgress.bind({ progress in
+            DispatchQueue.main.async { [weak self] in
+                self?.lblUploadProgress.text = "Uploading..."
```
Comment (Shilpriya-Simform) [inline UIComponents/UIComponents/WebServices/Module/Upload/Controller/MethodsViewController.swift:46]: use guard let with multple self in clouser

File: UIComponents/UIComponents/WebServices/Module/Upload/Controller/MethodsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                self?.lblUploadProgress.text = "\(Int(progress))%"
+            }
+        })
+
+    }
+    
+    // MARK: Open PickerView
+    private func openPickerView(_ sender: UIView) {
+        ImagePickerHelper.shared.pick(sender, viewController: self, completionHandler: { [weak self] (image, _) in
+            if let image {
```
Comment (Shilpriya-Simform) [inline UIComponents/UIComponents/WebServices/Module/Upload/Controller/MethodsViewController.swift:56]: use guard let

File: UIComponents/UIComponents/WebServices/ViewController/WebMainViewController.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    
+    // MARK: View Controller lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupViews()
+    }
+    
+    // MARK: SetUp Views
+    private func setupViews() {
+
```
Comment (Shilpriya-Simform) [inline UIComponents/UIComponents/WebServices/ViewController/WebMainViewController.swift:23]: if not in use method then we shoud remove from PR

File: UIComponents/UIComponents/WebServices/ViewController/WebMainViewController.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    
+    // MARK: SetUp Views
+    private func setupViews() {
+        
+    }
+    
+    // MARK: IBActions
+    @IBAction func btnNextTapped(_ sender: UIButton) {
+        Utility.changeVC(self, "News", NewsViewController.identifier)
```
Comment (Shilpriya-Simform) [inline UIComponents/UIComponents/WebServices/ViewController/WebMainViewController.swift:28]: why we use static string  "News"

**Review by Shilpriya-Simform — CHANGES_REQUESTED** [review]

File: UIComponents/UIComponents.xcodeproj/project.pbxproj (hunk 1 above)
Comment (SagarAjudiya) [reply UIComponents/UIComponents.xcodeproj/project.pbxproj:105]: OKay

**Pull request merged**